File: atcoder/abc042B_iroha_loves_strings.py
# Vicfred
# https://atcoder.jp/contests/abc042/tasks/abc042_b
# sorting, greedy
n, l = list(map(int, input().split()))

list_of_words = []

for _ in range(n):
    s = input()
    list_of_words.append(s)

list_of_words.sort()

answer = ""
for word in list_of_words:
    answer += word

print(answer)

File: atcoder/abc042B_iroha_loves_strings_.nim
# vicfred
# https://atcoder.jp/contests/abc042/tasks/abc042_b
# sorting

import algorithm, strutils, sequtils

let n = stdin.readLine.split[0].parseInt

var s = (1..n).mapIt(stdin.readLine)
s.sort system.cmp # system.cmp is required in 0.13.0

var ans = ""

for str in s:
  ans &= str

echo ans

